onnxruntime/core/providers/webnn/builders/impl/base_op_builder.cc

```diff
@@ -41,11 +41,12 @@ bool BaseOpBuilder::IsOpSupported(const InitializedTensorSet& initializers, cons
   return IsOpSupportedImpl(initializers, node, device_type, logger);
 }
 
-bool BaseOpBuilder::HasSupportedInputs(const Node& node, const emscripten::val& wnn_limits,
-                                       const logging::Logger& logger) const {
+bool BaseOpBuilder::HasSupportedInputs(const InitializedTensorSet& initializers, const Node& node,
+                                       const emscripten::val& wnn_limits, const logging::Logger& logger) const {
   const auto node_name = MakeString("Node [", node.Name(), "] type [", node.OpType(), "]");
   for (const auto* input : node.InputDefs()) {
-    if (!IsTensorShapeSupported(*input, node_name, logger)) {
+    // ONNX initializers should have shape information, skip the shape check if the input is an initializer.
```

Comment:
```suggestion
    // ONNX initializers should have shape information. Skip the shape check if the input is not an initializer.
```

Should it say "skip the shape check if the input is *not* an initializer"? Otherwise it doesn't appear to match "!Contains(initializers, input->Name()", and the comment is confusing because the line before it says "ONNX initializers should have shape information".

Reply:
Umm, that's indeed a bit confusing. I was intent to comment "Skip shape validation for initializers". How about:

> Skip the shape validation if the input is an initializer, as ONNX initializers are expected to have valid shape information.